### DISCUSSION TITLE
Will monaco-editor ever implement debug-adapter-protocol integration? [Feature Request]

### DISCUSSION THREAD

**Discussion by joecarl** [microsoft/monaco-editor#4073] [Q&A]
### Context

- [X] This issue is not a bug report. *(please use a different template for reporting a bug)*
- [X] This issue is not a duplicate of an existing issue. *(please use the [search](https://github.com/microsoft/monaco-editor/issues) to find existing issues)*

### Description

Will monaco-editor ever implement debug-adapter-protocol integration (DAP) as it does with LSP?

I mean, monaco currently implements functions like registerHoverProvider, registerSuggestionsProvider, etc. which allow you to integrate very well with LSP requests/responses. So will it ever be considered to implement analog functions for the debug adapter protocol? Something like: registerBreakpointProvider, registerWatchProvider or something similar.

I'm currently developing a custom IDE and had a great experience implementing the integrations with different language servers. So I wonder whether I will ever have the same experience implementing integrations with debuggers or I will have to dedicate more time to implement a custom function set for DAP.

**Comment by hediet** [comment]
Unfortunately, this is unlikely.
